### PR TITLE
ANDROAPP-5443-mobile-ui-Input-Shell

### PR DESCRIPTION
**Pull request by xavimolloy**

### CODE REVIEW

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/internal/ValueType.kt
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/internal/ValueType.kt:1]: where are you using this class

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/theme/TextColor.kt [hunk 1]
```diff
@@ -6,7 +6,7 @@ object TextColor {
     val OnSurface = Color.Charcoal800
     val OnSurfaceVariant = Color.Charcoal500
     val OnSurfaceLight = Color.Charcoal300
-    val OnError = Color.NeutralWhite
+    val OnError = Color.Red600
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/theme/TextColor.kt:9]: Are you sure, In the specs it shows NeutralWhite
image (5)

File: common/src/commonMain/kotlin/org/hisp/dhis/common/App.kt [hunk 2]
```diff
@@ -91,6 +100,10 @@ fun Main() {
 
         when (currentScreen.value) {
             Components.ICON_BUTTON -> IconButtonScreen()
+            Components.FORMS_COMPONENTS -> FormsComponentsScreen()
+
+            Components.FORMS_COMPONENTS -> FormsComponentsScreen()
+
             Components.RADIO -> RadioButtonScreen()
             else -> {}
```
[the remark this answers is not in the document]
Comment (xavimolloy) [reply common/src/commonMain/kotlin/org/hisp/dhis/common/App.kt:113]: This has been removed and the buttonStyle when statement was exhaustive but I have made this one too

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/internal/ValueType.kt
Comment (andresmr) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/internal/ValueType.kt:1]: I will delete it for know,, not sure if we should stick the components to the value types

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+            .width(Spacing.Spacing1)
+    )
+}
+
+/**
+ * DHIS2 InputShellIndicator, wraps Material 3 [Divider]
+ *  used in the [InputShell] component
+ */
+@Composable
+fun InputShellIndicator(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt:207]: shouldn't this be private?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        content()
+    }
+}
+
+/**
+ * DHIS2 InputShellButtonSeparator, wraps Material 3 [Divider]
+ * used in the [InputShell] component
+ */
+@Composable
+fun InputShellButtonSeparator(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt:190]: shouldn't his be private?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+/**
+ * DHIS2 InputShellRow, wraps Compose [Row]
+ * Row used in Input shell component for title, input field
+ * and buttons.
+ * @param backgroundColor controls the containers color
+ * @param content controls the content to be shown
+ */
+@Composable
+fun InputShellRow(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt:169]: Can this be private?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+ * @param title is the value of the text to be shown.
+ * @param state controls the  state of the component. Will be Unfocused by default
+ * @param primaryButton controls the primary button composable if null will show nothing
+ * @param secondaryButton controls  action button composable, if null will show nothing
+ * @param legend controls the optional legend composable
+ * @param inputField controls the input field composable .
+ * @param supportingText controls the supporting text composable
+ */
+@Composable
+fun InputShell(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt:96]: I think that I'll extract everything of Inputshell into a custom class called InputShell and split it from container. What do you think?

**Review by andresmr — CHANGES_REQUESTED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt (hunk 6 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt:96]: I thought about that and I agree it should be it's own separate component I'll extract it

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+import org.hisp.dhis.mobileui.designsystem.theme.Spacing
+import org.hisp.dhis.mobileui.designsystem.theme.SurfaceColor
+
+/**
+ * DHIS2 Text Input Field. Wraps Material· [BasicTextField].
+ * @param enabled Controls the enabled state of the component. When `false`, this component will not be
+ * clickable and will appear disabled to accessibility services.
+ */
+@Composable
+fun SimpleTextInputField(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt:32]: Do we want clients to use this component?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    )
+}
+
+/**
+ * DHIS2 EmptyInput
+ * empty input container with dashed border
+ * used for internal test purposes
+ */
+@Composable
+fun EmptyInput(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt:68]: Is this component going to be used by the final user?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputShell.kt [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+ * @param title is the value of the text to be shown.
+ * @param state controls the  state of the component. Will be Unfocused by default
+ * @param primaryButton controls the primary button composable if null will show nothing
+ * @param secondaryButton controls  action button composable, if null will show nothing
+ * @param legend controls the optional legend composable
+ * @param inputField controls the input field composable .
+ * @param supportingText controls the supporting text composable
+ */
+@Composable
+fun InputShell(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputShell.kt:41]: Is this going to be used by the client?. Shouldn't be internal?

**Review by andresmr — COMMENTED** [review]
I added some comments about methods privacy

**Review by andresmr — APPROVED** [review]

File: common/src/commonMain/kotlin/org/hisp/dhis/common/screens/IconButtonScreen.kt [hunk 10]
```diff
@@ -1,87 +1,73 @@
...
-                            IconButtonPreview(true, IconButtonStyle.FILLED)
-                            IconButtonPreview(false, IconButtonStyle.FILLED)
-                        }
+            )
+
+            RowComponentContainer(
+                title = "Filled",
+                content = {
+                    Row(
+                        horizontalArrangement = Arrangement.spacedBy(10.dp)
```
Comment (DavidAparicioAlbaAsenjo) [inline common/src/commonMain/kotlin/org/hisp/dhis/common/screens/IconButtonScreen.kt:42]: try to use constant values instead

File: common/src/commonMain/kotlin/org/hisp/dhis/common/screens/ButtonScreen.kt [hunk 11]
```diff
@@ -1,127 +1,71 @@
...
-                ) {
-                    ButtonPreviewWithIcon("Label", ButtonStyle.FILLED)
-                    ButtonPreviewWithIcon("Label", ButtonStyle.FILLED, false)
-                }
-            }
-        )
-
+    ColumnComponentContainer("Buttons") {
+        Text("Filled")
+        RowComponentContainer() {
```
Comment (DavidAparicioAlbaAsenjo) [inline common/src/commonMain/kotlin/org/hisp/dhis/common/screens/ButtonScreen.kt:15]: you can remove unnecesary parentheses

**Review by DavidAparicioAlbaAsenjo — APPROVED** [review]

**Pull request merged**